src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs

```diff
@@ -122,6 +122,28 @@ public static string SystemDirectory
             }
         }
 
+        public static unsafe bool UserInteractive
+        {
+            get
+            {
+                // Per documentation of GetProcessWindowStation, this handle should not be closed
+                IntPtr handle = Interop.User32.GetProcessWindowStation();
+                if (handle != IntPtr.Zero)
+                {
+                    Interop.User32.USEROBJECTFLAGS flags = default;
+                    uint dummy = 0;
+                    if (Interop.User32.GetUserObjectInformationW(handle, Interop.User32.UOI_FLAGS, &flags, (uint)sizeof(Interop.User32.USEROBJECTFLAGS), ref dummy))
+                    {
+                        return ((flags.dwFlags & Interop.User32.WSF_VISIBLE) == 0);
```

Comment:
It looks like you forgot to invert this logic.   Also, should we add back the static property like we had in desktop?
https://referencesource.microsoft.com/#mscorlib/system/environment.cs,1435

Reply:
Ack - thank you @ericstj !

Re the static, @jkotas pointed out that there may be a way to change a process to/from interactive - although I didn't do it successfully myself. See mention here https://github.com/dotnet/runtime/issues/770#issuecomment-570006411 I saw no reason to cache the value.

Reply:
Ok, I'll make the simple fix for now.  FWIW I'm now able to reproduce the test failure, so I can confirm fail/then-fix.

Reply:
That would explain https://github.com/dotnet/runtime/issues/1724 also. I was not albe to run the service controller tests locally because of #1262. I tested locally by hand instead, and apparently did so before I made this mistake.

Reply:
Comments crossed. Great, I was about to do it, but apparently you've already got those tests working so you can verify it as well. thanks